### PR TITLE
docs: revisit and clarify some parts SL-1879

### PR DESCRIPTION
**Pull request by XVincentX**
* Fix some imprecise parts (we're not using styled-components, for example)
* Remove the component list (it was outdated and not providing any useful help)
* Use embedded markdown links where possible
* Add a FAQ about what are the "strange" properties (this is what would have helped me)
* Put the migration guide on bottom
* Put the Creating a Component section on the bottom since it's not the default use case

### CODE REVIEW

**Comment by XVincentX** [comment]
Marc, Nick and Chris added as reviewers.

**Comment by XVincentX** [comment]
Maybe we can also use @philsturgeon's help on this.

File: README.md [hunk 1]
```diff
@@ -96,6 +65,17 @@ IconLibrary.add(faCaretDown);
   - [Git Repo](https://github.com/jxnblk/styled-system)
   - [Site](https://jxnblk.com/styled-system/)
 
+### Migration guide
+
+In case you're porting a component from another Stoplight's Library [see the this document for some guidance](./docs/migration.md)
+
+### Anti panic Faq
```
Comment (wmhilton) [inline README.md]:
```suggestion
### FAQ
```

Comment (XVincentX) [reply README.md]: You always kill all the fun.

**Comment by philsturgeon** [comment]
These are good changes, and I don't have anything else to say :D

**Pull request merged**